soundata/datasets/fsd50k.py

```diff
@@ -0,0 +1,478 @@
+            audioset_to_fsd50k[line[2]] = line[1]
+
+    return fsd50k_to_audioset, audioset_to_fsd50k
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The FSD50K dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="fsd50k",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
+
+    @core.copy_docs(load_ground_truth)
+    def load_ground_truth(self, *args, **kwargs):
+        return load_ground_truth(*args, **kwargs)
+
+    @core.copy_docs(load_fsd50k_vocabulary)
+    def load_fsd50k_vocabulary(self, *args, **kwargs):
+        return load_fsd50k_vocabulary(*args, **kwargs)
+
+    @property
+    def fsd50k_to_audioset(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "vocabulary.csv"
+        )
+        return load_fsd50k_vocabulary(data_path)[0]
+
+    @property
+    def audioset_to_fsd50k(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "vocabulary.csv"
+        )
+        return load_fsd50k_vocabulary(data_path)[1]
+
+    @property
+    def class_info(self):
+        data_path = os.path.join(
+            self.data_home, "FSD50K.metadata", "class_info_FSD50K.json"
+        )
+        return json.load(open(data_path, "r")) if os.path.exists(data_path) else None
+
+    @core.cached_property
+    def _metadata(self):
+        # Ground_truth path
+        ground_truth_dev_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "dev.csv"
+        )
+        ground_truth_eval_path = os.path.join(
+            self.data_home, "FSD50K.ground_truth", "eval.csv"
+        )
+
+        # Load clip metadata path
+        clips_info_dev_path = os.path.join(
+            self.data_home, "FSD50K.metadata", "dev_clips_info_FSD50K.json"
+        )
+        clips_info_eval_path = os.path.join(
+            self.data_home, "FSD50K.metadata", "eval_clips_info_FSD50K.json"
+        )
+
+        # Load PP/PNP ratings
+        pp_pnp_ratings_path = os.path.join(
+            self.data_home, "FSD50K.metadata", "pp_pnp_ratings_FSD50K.json"
+        )
+
+        metadata_index = {
+            "ground_truth_dev": load_ground_truth(ground_truth_dev_path),
+            "ground_truth_eval": load_ground_truth(ground_truth_eval_path),
+            "clips_info_dev": json.load(open(clips_info_dev_path, "r"))
+            if os.path.exists(clips_info_dev_path)
+            else None,
+            "clips_info_eval": json.load(open(clips_info_eval_path, "r"))
+            if os.path.exists(clips_info_eval_path)
+            else None,
+            "pp_pnp_ratings": json.load(open(pp_pnp_ratings_path, "r"))
+            if os.path.exists(pp_pnp_ratings_path)
```

Comment:
I'm a bit confused about the structure of the metadata in this dataset. I understand that you have different ground truth depending if it's `dev` or `eval`? What is `clips_info` and `pp_pnp_ratings`? 

I guess what I'm wondering is if we can get this to the standard format of dataset metadata which is having the `clip_ids` as `keys`? What do you think @genisplaja?

Reply:
About the ground truth: Yes, `dev` and `eval` are annotated in different files and in a slightly different format, although I adapted the loading function so we can load both using a single `load_ground_truth` function (learning the lesson from mirdata...!). `clips_info` is the Freesound metadata of the clip, and `pp_pnp_ratings` are at clip-level and there are PP and PNP ratings for each label of a clip (detailed info about calculation and meaning of these ratings is in Section 3 of the paper: [https://arxiv.org/pdf/2010.00475.pdf](https://arxiv.org/pdf/2010.00475.pdf)).

The thing is, the dataset files are general for both annotations and metadata. I guess that to have the regular metadata format we would need different dataset-level dicts?

Reply:
Done:) You can ignore my last comment on this thread!! I already found a solution that seems to work fine. It is already committed.